src/libsystemd/sd-json/json-util.c

```diff
@@ -281,3 +282,63 @@ int json_dispatch_pidref(const char *name, sd_json_variant *variant, sd_json_dis
 
         return 0;
 }
+
+static int json_variant_new_stat(sd_json_variant **ret, struct stat *st) {
+        char mode[64];
+
+        assert(st);
+
+        if (!stat_is_set(st))
+                return sd_json_variant_new_null(ret);
+
+        xsprintf(mode, "%04o", st->st_mode & ~S_IFMT);
+
+        return sd_json_buildo(
+                        ret,
+                        SD_JSON_BUILD_PAIR(
+                                        "dev",
+                                        SD_JSON_BUILD_OBJECT(
+                                                        SD_JSON_BUILD_PAIR_UNSIGNED("major", major(st->st_dev)),
+                                                        SD_JSON_BUILD_PAIR_UNSIGNED("minor", minor(st->st_dev)))),
```

Comment:
maybe add a helper json_variant_new_devno() or so, which takes a dev_t and formats this?

and then a JSON_BUILD_DEVNO() helper along with it?